test/test_interface_ixxat.py

```diff
@@ -0,0 +1,63 @@
+"""
+Unittest for ixxat interface.
+
+Copyright (C) 2019 Marcel Kanter <marcel.kanter@googlemail.com>
+"""
+
+import unittest
+import can
+
+from can import CanError, CanBackEndError, CanInitializationError, CanOperationError
+
+class SoftwareTestCase(unittest.TestCase):
+    """
+    Test cases that test the software only and do not rely on an existing/connected hardware.
+    """
+    def setUp(self):
+        pass
+    
+    def tearDown(self):
+        pass
+    
+    def test_bus_creation(self):
+        # channel must be >= 0
+        with self.assertRaises(ValueError):
+            bus = can.Bus(interface = 'ixxat', channel = -1)
+        
+        # rxFifoSize must be > 0
+        with self.assertRaises(ValueError):
+            bus = can.Bus(interface = 'ixxat', channel = 0, rxFifoSize = 0)
+        
+        # txFifoSize must be > 0
+        with self.assertRaises(ValueError):
+            bus = can.Bus(interface = 'ixxat', channel = 0, txFifoSize = 0)
+
+
+class HardwareTestCase(unittest.TestCase):
+    """
+    Test cases that rely on an existing/connected hardware.
+    """
+    def setUp(self):
+        try:
+            bus = can.Bus(interface = 'ixxat', channel = 0)

```

Comment:
Also close the bus afterwards?

Reply:
Fixed.

Reply:
Well, I wanted to say that this is required in all cases, so also in the actual tests. In general you could use the bus as a context manager:

```
with can.Bus(...) as bus:
    bus.send(whatever)
```

This does not require a try/finally block.